ratatui-widgets/src/reflow.rs

```diff
@@ -8,10 +8,14 @@ use unicode_width::UnicodeWidthStr;
 /// A state machine to pack styled symbols into lines.
 /// Cannot implement it as Iterator since it yields slices of the internal buffer (need streaming
 /// iterators for that).
-pub trait LineComposer<'a> {
+pub trait StatefulLineComposer<'a> {
     fn next_line<'lend>(&'lend mut self) -> Option<WrappedLine<'lend, 'a>>;
 }
 
+pub trait LineComposer<'a> {
+    fn next_line(&mut self) -> Option<WrappedLine<'_, 'a>>;
+}
+
```

Comment:
If there's only one impl of each of these, then it's likely not worth having a trait. The trait only existed to make the shared behavior accessible via the same methods. What would this look like without the traits?